https://github.com/mgsloan/store/pull/159

From 937b8f1cec39ceb91e37d5b2e0f815c13cb0d4d9 Mon Sep 17 00:00:00 2001
From: Sergei Trofimovich <slyfox@gentoo.org>
Date: Sun, 18 Apr 2021 13:27:47 +0100
Subject: [PATCH] test/Data/StoreSpec.hs: tweak for network-3.1
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

On networt-3.1 testsuite fails build on duplicate instance definition
for Bounded PortNumber:

```
test/Data/StoreSpec.hs:105:10: error:
    Duplicate instance declarations:
      instance Bounded PortNumber
        -- Defined at test/Data/StoreSpec.hs:105:10
      instance Bounded PortNumber
        -- Defined in ‘network-3.1.2.1:Network.Socket.Types’
```

The change uses network's instance if available.

Signed-off-by: Sergei Trofimovich <slyfox@gentoo.org>
---
 test/Data/StoreSpec.hs | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/test/Data/StoreSpec.hs b/test/Data/StoreSpec.hs
index a09047f..34304a2 100644
--- a/test/Data/StoreSpec.hs
+++ b/test/Data/StoreSpec.hs
@@ -102,9 +102,11 @@ $(mkManyHasTypeHash [ [t| Int32 |] ])
 -- Serial instances for (Num a, Bounded a) types. Only really
 -- appropriate for the use here.
 
+#if !MIN_VERSION_network(3,1,0)
 instance Bounded PortNumber where
   minBound = 0
   maxBound = 65535
+#endif
 
 $(do let ns = [ ''PortNumber
 
-- 
2.31.1

